File: src/main/java/kr/co/bookand/backend/common/domain/Status.java
package kr.co.bookand.backend.common.domain;

public enum Status {
    INVISIBLE, VISIBLE, REMOVE;

    public static Status toEnum(String status) {
        if (status.equals("VISIBLE")) {
            return VISIBLE;
        } else if (status.equals("INVISIBLE")) {
            return INVISIBLE;
        } else if (status.equals("REMOVE")) {
            return REMOVE;
        } else {
            throw new IllegalArgumentException("잘못된 상태값입니다.");
        }
    }
}

File: src/main/kotlin/kr/co/bookand/backend/common/Status.kt
package kr.co.bookand.backend.common

import kr.co.bookand.backend.common.exception.BookandException


enum class Status {
    INVISIBLE, VISIBLE, REMOVE;

    companion object {
        fun toEnum(status: String): Status {
            return when (status) {
                INVISIBLE.name -> INVISIBLE
                VISIBLE.name -> VISIBLE
                REMOVE.name -> REMOVE
                else -> throw BookandException(ErrorCode.INPUT_VALID_ERROR)
            }
        }
    }
}
